ambition_utils/rrule/models.py

```diff
@@ -330,16 +330,20 @@ def generate_dates(self, num_dates=20):
             # Convert to time zone
             date_with_tz = fleming.attach_tz_if_none(d, self.time_zone)
             date_in_utc = fleming.convert_to_tz(date_with_tz, pytz.utc, True)
-            dates.append(date_in_utc)
-
-            for x in range(0, num_dates):
+            
+            if not start_date or date_in_utc > start_date:
+                dates.append(date_in_utc)
+            
+            while len(dates) < num_dates:
                 d = rule.after(d)
                 if not d:
                     break
                 # Convert to time zone
                 date_with_tz = fleming.attach_tz_if_none(d, self.time_zone)
                 date_in_utc = fleming.convert_to_tz(date_with_tz, pytz.utc, True)
-                dates.append(date_in_utc)
+
+                if not start_date or date_in_utc > start_date:
```

Comment:
would it be easier to just do one check at the start of the method to make sure the `start_date` is within a certain range? if start date exists and is < this date, then it will pass every check for future dates.

Reply:
would it work the same if you use the passed `start_date` instead of `d = rule[0]` for the first date?

Reply:
I improved this chunk. Tell me what you think. 7d85586

> would it be easier to just do one check

As it was written, `date_in_utc` was not the same inside and outside of the loop. I refactored this into a method to share between the initial `append` and loop.

> would it work the same if you use the passed start_date instead of d = rule[0] for the first date?

That angle works for the loop but `start_date` may not be a date in the series so we cannot append that date as the first of the series.